compiler/rustc_ast_lowering/src/asm.rs

```diff
@@ -368,7 +371,7 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
 
         // If a clobber_abi is specified, add the necessary clobbers to the
         // operands list.
-        if let Some((abi, abi_span)) = clobber_abi {
+        for (abi, abi_span) in clobber_abis {
```

Comment:
Make sure you don't emit duplicate `Out` operands for the same register from multiple clobbers.